pkg/commands/compute/init.go

```diff
@@ -392,15 +415,113 @@ func packageAuthors(authors []string, manifestEmail string, in io.Reader, out io
 	return authors, nil
 }
 
+// configureJsToolchain validates the JS toolchain has been configured
+// correctly and will take the following priority order:
+//
+// - Flag
+// - Environment
+// - File
+// - Default
+func configureJsToolchain(
+	toolchainFlag, configPath string,
+	toolchains []string,
+	f config.File,
+	acceptDefaults bool,
+	in io.Reader, out io.Writer,
+) string {
+	toolchain, source := f.JsToolchain()
+
+	if toolchainFlag != "" {
+		toolchain = toolchainFlag
+		source = config.SourceFlag
+	}
+
+	defaultMessage := fmt.Sprintf("Toolchain option unrecognised. Defaulting to %s", config.ToolchainJsDefault)
+	validator := validateToolchainOption(toolchains)
+
+	switch source {
+	case config.SourceDefault:
+		// Before accepting default toolchain, prompt the user for their choice.
+		if !acceptDefaults {
```

Comment:
Could we perhaps check for the existence of `package-lock.json` (which indicates npm) or `yarn.lock` (which indicates yarn), rather than requiring the user to choose? This would mean that `fastly compute publish` would work when pulling down a template/project that uses Yarn.